src/components/Dropdown/PrioritySelector/PrioritySelector.tsx

```diff
@@ -0,0 +1,111 @@
+import { IconButton } from '../../Buttons';
+
+const iconMap = {
+    mandatory: KeyboardDoubleArrowUp,
+    important: KeyboardArrowUp,
+    optional: KeyboardArrowDown,
+    exclude: Close,
+};
+
+export type Priority = 'mandatory' | 'important' | 'optional' | 'exclude';
+
+export type PriorityItem = {
+    priority: Priority;
+    label: string;
+    value?: string;
+};
+
+const defaultList: PriorityItem[] = [
+    { priority: 'mandatory', label: 'Mandatory' },
+    { priority: 'important', label: 'Important' },
+    { priority: 'optional', label: 'Optional' },
+    { priority: 'exclude', label: 'Exclude' },
+];
+
+export interface Props<>extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'> {
+    /** Currently selected priority item that indicates the importance of the component. */
+    selectedItem: PriorityItem;
+    /** Array of availible priority items. */
+    list: PriorityItem[];
+    /** Callback function triggered when a new priority is selected. */
+    onChange: (newPriorityItem: PriorityItem) => void;
+    /** Boolean indicating whether the whole badge should be disabled. */
+    isDisabled?: boolean;
+    /** Priority button label name for ARIA labelling */
+    buttonLabel?: string;
+}
+
+const { block, elem } = bem('PrioritySelector', styles);
+export function PrioritySelector({
+    selectedItem,
+    isDisabled = false,
+    onChange,
+    buttonLabel,
+    list = defaultList,
+    ...rest
+}: Props) {
+    const badgeRef = React.useRef<HTMLDivElement | null>(null);
+
+    const hasPriorityList = list && list.length > 0;
+
+    const renderPriorityIcon = (priorityType?: Priority, disabled: boolean = false) => {
+        if (!priorityType) {
+            return null;
+        }
+
+        const IconComponent = iconMap[priorityType];
+
+        return IconComponent ? (
+            <IconComponent
+                {...elem('icon', { [priorityType]: true })}
+                disabled={disabled}
+                viewBox="0 0 24 24"
+            />
+        ) : null;
+    };
+
+    return (
+        <div {...block({ ...rest })} ref={badgeRef}>
+            {hasPriorityList && (
+                <Dropdown<PriorityItem>
+                    button={
+                        <IconButton
+                            {...elem('icon', { [selectedItem.priority]: true })}
+                            variant="ghost"
+                            aria-label={buttonLabel}
+                            disabled={isDisabled}
+                            type="button"
+                        >
+                            {renderPriorityIcon(selectedItem.priority, isDisabled)}
+                        </IconButton>
+                    }
+                    onChange={(newPriorityItem) => onChange(newPriorityItem)}
+                    placement="bottom-start"
+                    listClassName={styles.badgeDropdownList}
+                    refElement={badgeRef}
+                >
+                    {list.map((item) => (
```

Comment:
I wonder if this can cause a problem if the order is wrong, and if we should control for that or not?
It means looping on Priority, then looking up the correct PriorityItem from the list and rendering that. For that it would be easier if this was an object like thing, something like `{key[Priority]: PriorityItem}` even thought that is a bit of a duplication. What do you think?

BTW: if you do this, you could make `Priority` and Enum, so that you can use it both in JS and TS. (Cannot remember if we have that here in OneUI or only in JF if you want to see examples)